### PR TITLE
 Longest Increasing Subsequence code

### LINKED ISSUE #528: Longest Increasing Subsequence
Hey,
I am a GSSOC'21 participant, I would like to add a problem - longest increasing subsequence in an array in DP section in C++
@tarun26091999 Can you please assign this issue to me.

Thank you

### PR DESCRIPTION
**Pull request by Chirag018**
Fixes #528 
Adding LIS code in java

### CODE REVIEW

File: Algorithm/DP/longest_increasing_subsequence.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+                if (my_arr[i] > my_arr[j] && seq_arr[i] < seq_arr[j] + 1)
+                    seq_arr[i] = seq_arr[j] + 1;
+        for (i = 0; i < arr_len; i++)
+            if (max < seq_arr[i])
+                max = seq_arr[i];
+        return max;
+    }
+
+    public static void main(String args[]) {
+        int my_arr[] = { 10,9,2,5,3,7,101,18 };
```
Comment (raksha009) [inline Algorithm/DP/longest_increasing_subsequence.java:34]: Take input from user.

File: Algorithm/DP/longest_increasing_subsequence.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        int arr_len = my_arr.length;
+        System.out.println("The length of the longest increasing subsequence is " + incre_subseq(my_arr, arr_len));
+    }
+}
+
+// Time Complexity is O(n^2)
+// Space Complexity is O(n)
```
Comment (raksha009) [inline Algorithm/DP/longest_increasing_subsequence.java:41]: Specify what is n

File: Algorithm/DP/longest_increasing_subsequence.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        return max;
+    }
+
+    public static void main(String args[]) {
+        int my_arr[] = { 10,9,2,5,3,7,101,18 };
+        int arr_len = my_arr.length;
+        System.out.println("The length of the longest increasing subsequence is " + incre_subseq(my_arr, arr_len));
+    }
+}
+
```
Comment (raksha009) [inline Algorithm/DP/longest_increasing_subsequence.java:39]: Add 2 test cases here.

**Review by raksha009 — CHANGES_REQUESTED** [review]

**Review by Chirag018 — COMMENTED** [review]
Updated Successfully

**Comment by raksha009** [comment]
Plz go through the guidelines and link yr code with respective folder.

**Comment by Chirag018** [comment]
Okay

**Comment by raksha009** [comment]
Plz close this pr if its a duplicate one.

**Pull request closed without merging**